### DISCUSSION TITLE
How do you redirect to the 404 page from within getServerSideProps()?

### DISCUSSION THREAD

**Discussion by ahmadjonhn** [vercel/next.js#17652] [Help]
export async function getServerSideProps({ params }) {
  
   const res = await axios.get(`${url}/api/categories?slug=${params.id}`)
   const data = await res.data.data;
   console.log(data)
   // Pass data to the page via props
   return { props: { data } };
}

How can I redirect to my 404.js page in this async function? @timneutkens

**Comment by ahmadjonhn** [comment]
When I try to type not existing slug, response is undefined, I can not get the idea

**Reply by ahmadjonhn** [reply]
axios.interceptors.response.use((response) => {

  const {status, data, statusText} = response;
  if (!status) {
      return connectionError('Ошибка соединения!', this);
  }
  return response;

}, ({response}) => {
  const {status, data, statusText} = response;
  const cookies = new Cookies();

**Reply by ahmadjonhn** [reply]
Server Error
TypeError: Cannot destructure property 'status' of 'response' as it is undefined.


And the server Error is received so status in my axios interceptor can't be destructured

**Comment by dolie** [comment]
This looks like it is a problem with Axios, not with Next...  
Errors are handled natively. If you get `404` with `await axios.get(${url}/api/categories?slug=${params.id})`, Next will throw it to client. Client will reroute to `404.jsx`.  
The thing is that it is only visible in production mode.
In dev mode, you will see the error stack.
